src/Websites/Websites.Test/ScenarioTests/CertificatesTests.ps1

```diff
@@ -0,0 +1,142 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Tests creating a new managed cert for app.
+#>
+function Test-NewAzWebAppManagedCertificate
+{
+	$rgname = "RG-W-CUS"
+	$appname = "managedcerts"
+	$slot = "testslot"
+	$prodHostname = "www.managedcerts.net"
+	$slotHostname = "testslot.adorenow.net"
+	$thumbprint=""
+
+	try{		
+
+		$cert=New-AzWebAppManagedCertificate -ResourceGroupName $rgname -WebAppName $appname -HostName $prodHostname
+		$thumbprint=$cert.ThumbPrint
+
+		# Assert
+		Assert-AreEqual $prodHostname $cert.SubjectName		
+	
+	}
+	finally{
+		
+		# Cleanup
+		Remove-AzWebAppManagedCertificate -ResourceGroupName $rgname -WebAppName $appname -HostName $prodHostname -ThumbPrint $thumbprint
+	}
+}
+
+<#
+.SYNOPSIS
+Tests creating a new managed cert for app with SSL binding.
+#>
+function Test-NewAzWebAppManagedCertificateWithSSLBinding
+{
+	$rgname = "RG-W-CUS"
+	$appname = "managedcerts"
+	$slot = "testslot"
+	$prodHostname = "www.managedcerts.net"
+	$slotHostname = "testslot.adorenow.net"
+	$thumbprint=""
+
+	try{		
+
+		$cert=New-AzWebAppManagedCertificate -ResourceGroupName $rgname -WebAppName $appname -HostName $prodHostname -AddCertBinding
+		$thumbprint=$cert.ThumbPrint
+
+		# Assert
+		Assert-AreEqual $prodHostname $cert.SubjectName		
+
+		#Assert
+		$getResult = Get-AzWebAppSSLBinding -ResourceGroupName $rgname -WebAppName  $appname
+		Assert-AreEqual 1 $getResult.Count
+		$currentHostNames = $getResult | Select -expand Name
+		Assert-True { $currentHostNames -contains $prodHostname }
+		$getResult = Get-AzWebAppSSLBinding -ResourceGroupName $rgname -WebAppName  $appname -Name $prodHostname
+		Assert-AreEqual $getResult.Name $prodHostname
+	
+	}
+	finally{
+		
+		# Cleanup
+		Remove-AzWebAppSSLBinding -ResourceGroupName $rgname -WebAppName  $appname -Name $prodHostname -Force
+		Remove-AzWebAppManagedCertificate -ResourceGroupName $rgname -WebAppName $appname -HostName $prodHostname -ThumbPrint $thumbprint
+	}
+}
+
+<#
+.SYNOPSIS
+Tests creating a new managed certfor slot.
+#>
+function Test-NewAzWebAppManagedCertificateForSlot
+{
+
+	$rgname = "RG-W-CUS"
```

Comment:
Can we change the test to create a webapp & create managed cert for default hostname or create a traffic manger end point using Powershell & then add managed cert for this traffic manager endpoint? Basically make the test not need to have a app & if possible hostname to exist. Thanks!

Reply:
> Can we change the test to create a webapp & create managed cert for default hostname or create a traffic manger end point using Powershell & then add managed cert for this traffic manager endpoint? Basically make the test not need to have a app & if possible hostname to exist. Thanks!

facing some challenges in writing tests for traffic manager and it is taking time. will update tests before we merge these changes to master.